### PR TITLE
Enable resource locking

### PR DESCRIPTION
**Pull request by avi-hmcts**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: scripts/disable-resource-locking.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+set -e
+
+SUBS=$1
+
+echo "retrieve all resource groups in a subscription"
+RESOURCE_GROUPS=$(az group list | jq '.[].name')
```
Comment (naikajah) [inline scripts/disable-resource-locking.sh:7]: Do not put lock on all the resources. Just put locks first on Database/Keyvault/app insights

File: scripts/enable-resource-locking.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+set -e
+
+SUBS=$1
+
+echo "retrieve all resource groups in a subscription"
+RESOURCE_GROUPS=$(az group list | jq '.[].name')
```
Comment (naikajah) [inline scripts/enable-resource-locking.sh:7]: Same here. Should not be for all the resource.

File: scripts/enable-resource-locking.sh [hunk 3]
```diff
@@ -4,13 +4,16 @@ set -e
 SUBS=$1
 
 echo "retrieve all resource groups in a subscription"
-RESOURCE_GROUPS=$(az group list | jq '.[].name')
+RESOURCE_GROUPS=$(az group list --subscription DCD-CFTAPPS-SBOX | jq '.[].name')
 
 for rg in ${RESOURCE_GROUPS[@]}; 
 do
   RG_NAME=$(sed -e 's/^"//' -e 's/"$//' <<<"$rg")
+  RSRC_LIST=$(az resource list -g $RG_NAME --subscription DCD-CFTAPPS-SBOX | jq '.[].type' | sed -e 's/^"//' -e 's/"$//' | grep -ciE 'Storage|KeyVault|Sql')
```
Comment (timja) [inline scripts/enable-resource-locking.sh:12]: can't you use a query rather than jq + sed + grep?

Comment (avi-hmcts) [reply scripts/enable-resource-locking.sh]: added a JSONPATH for retrieving the resource group list enforcing locks.

File: scripts/enable-resource-locking.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+JSONPATH="[?(contains(type, 'Storage') || contains(type, 'KeyVault') || contains(type, 'SQL') || contains(type, 'Insights')) ].[resourceGroup]"
+
+echo "retrieve all resource groups in a subscription"
+RG_LIST=$(az resource list --query "${JSONPATH}"  -o tsv | sort -u)
+
+
+for rg in ${RG_LIST[@]}
+do
+  RG_NAME=$(sed -e 's/^"//' -e 's/"$//' <<<"$rg")
```
Comment (thomast1906) [inline scripts/enable-resource-locking.sh:12]: As discussed on Slack - Can you check if $RG_NAME needed? Should be able to just use $rg

**Review by thomast1906 — CHANGES_REQUESTED** [review]

File: scripts/enable-resource-locking.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  locks=$(az lock list -g $rg -o tsv)
+  
+  subname=$(az account show | jq .name)
+  echo "sub: $subname"
+  if [[ $subname == *"STG"* ]]
+  then
+    lockname="stg-lock"
+  else
+    lockname="prod-lock"
+  echo $lockname
```
Comment (thomast1906) [inline scripts/enable-resource-locking.sh:22]: Can you remove?

**Review by thomast1906 — CHANGES_REQUESTED** [review]
Remove reference of:-
  echo $lockname

**Review by thomast1906 — APPROVED** [review]

**Pull request merged**